src/java.base/linux/classes/sun/nio/ch/EPollSelectorImpl.java

```diff
@@ -125,7 +125,7 @@ protected int doSelect(Consumer<SelectionKey> action, long timeout)
                 if (numEntries == IOStatus.INTERRUPTED && timedPoll) {
                     // timed poll interrupted so need to adjust timeout
                     long adjust = System.nanoTime() - startTime;
-                    to -= TimeUnit.MILLISECONDS.convert(adjust, TimeUnit.NANOSECONDS);
+                    to -= (int)TimeUnit.MILLISECONDS.convert(adjust, TimeUnit.NANOSECONDS);
```

Comment:
Can we change this `to =- (int) TimeUnit.NANOSECONDS.toMillis(adjust);` while we here?